src/coreclr/vm/jitinterface.cpp

```diff
@@ -11685,43 +11685,77 @@ bool CEEInfo::getStaticFieldContent(CORINFO_FIELD_HANDLE fieldHnd, uint8_t* buff
+                            const unsigned gcSlotBegin = i * TARGET_POINTER_SIZE;
+                            const unsigned gcSlotEnd = gcSlotBegin + TARGET_POINTER_SIZE;
 
-                                if (subFieldValue == nullptr)
-                                {
-                                    // Report null
-                                    memset(buffer, 0, bufferSize);
-                                    result = true;
-                                }
-                                else if (GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(subFieldValue))
-                                {
-                                    CORINFO_OBJECT_HANDLE handle = getJitHandleForObject(
-                                        ObjectToOBJECTREF(subFieldValue), /*knownFrozen*/ true);
+                            if (gcSlotBegin >= (unsigned)valueOffset && gcSlotEnd <= (unsigned)(valueOffset + bufferSize))
+                            {
+                                // GC slot intersects with our valueOffset + bufferSize - we can't use memcpy...
+                                canUseMemcpy = false;
 
-                                    // GC handle is either from FOH or null
-                                    memcpy(buffer, &handle, bufferSize);
-                                    result = true;
+                                // ...unless we're interested in that GC slot's value itself
+                                if (gcSlotBegin == (unsigned)valueOffset && gcSlotEnd == (unsigned)(valueOffset + bufferSize) && ptr[i] == TYPE_GC_REF)
+                                {
+                                    _ASSERT((UINT)bufferSize == sizeof(CORINFO_OBJECT_HANDLE));
+
+                                    GCX_COOP();
+
+                                    // Read field's value under COOP mode
+                                    Object* gcSlotValue = nullptr;
+                                    memcpy(&gcSlotValue, (uint8_t*)baseAddr + gcSlotBegin, sizeof(CORINFO_OBJECT_HANDLE));
+
+                                    if (gcSlotValue == nullptr)
+                                    {
+                                        // GC handle is null
+                                        memset(buffer, 0, bufferSize);
+                                        result = true;
+                                    }
+                                    else if (GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(gcSlotValue))
+                                    {
+                                        // GC handle is a frozen (nongc) object
+                                        CORINFO_OBJECT_HANDLE handle = getJitHandleForObject(ObjectToOBJECTREF(gcSlotValue), /*knownFrozen*/ true);
+                                        memcpy(buffer, &handle, bufferSize);
+                                        result = true;
+                                    }
```

Comment:
```suggestion
                                    result = getStaticObjRefContent(ObjectToOBJECTREF(*(Object**)((uint8_t*)baseAddr + gcSlotBegin)), buffer, ignoreMovableObjects);
```
+ factor this out `getStaticObjRefContent` helper method and call this helper method in the other place that returns object reference from this method.

Reply:
Thanks! Will do